test/common/creation.go

```diff
@@ -0,0 +1,248 @@
+var coreAPIVersion = corev1.SchemeGroupVersion.Version
+var rbacAPIGroup = rbacv1.SchemeGroupVersion.Group
+var rbacAPIVersion = rbacv1.SchemeGroupVersion.Version
+
+// CreateChannel will create a Channel Resource in Eventing.
+func (client *Client) CreateChannel(name, provisonerName string) error {
+	namespace := client.Namespace
+	provisioner := base.ClusterChannelProvisioner(provisonerName)
+	channel := base.Channel(name, namespace, provisioner)
+	channels := client.Eventing.EventingV1alpha1().Channels(namespace)
+
+	_, err := channels.Create(channel)
+	if err != nil {
+		return err
+	}
+	client.Cleaner.Add(eventingAPIGroup, eventingAPIVersion, "channels", namespace, name)
+	return nil
+}
+
+// CreateChannels will create a list of Channel Resources in Eventing.
+func (client *Client) CreateChannels(names []string, provisionerName string) error {
+	for _, name := range names {
+		if err := client.CreateChannel(name, provisionerName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CreateSubscription will create a Subscription.
+func (client *Client) CreateSubscription(name, channelName, subscriberName, replyName string) error {
+	namespace := client.Namespace
+	channel := base.ChannelRef(channelName)
+	subscriber := base.SubscriberSpecForService(subscriberName)
+	reply := base.ReplyStrategyForChannel(replyName)
+	subscription := base.Subscription(name, namespace, channel, subscriber, reply)
+	subscriptions := client.Eventing.EventingV1alpha1().Subscriptions(namespace)
+	_, err := subscriptions.Create(subscription)
+	if err != nil {
+		return err
+	}
+	client.Cleaner.Add(eventingAPIGroup, eventingAPIVersion, "subscriptions", namespace, name)
+	return nil
+}
+
+// CreateSubscriptions will create a list of Subscriptions.
+func (client *Client) CreateSubscriptions(names []string, channelName, subscriberName, replyName string) error {
+	for _, name := range names {
+		if err := client.CreateSubscription(name, channelName, subscriberName, replyName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CreateBroker will create a Broker.
+func (client *Client) CreateBroker(name, provisionerName string) error {
+	namespace := client.Namespace
+	provisioner := base.ClusterChannelProvisioner(provisionerName)
+	broker := base.Broker(name, namespace, provisioner)
+	brokers := client.Eventing.EventingV1alpha1().Brokers(namespace)
+	_, err := brokers.Create(broker)
+	if err != nil {
+		return err
+	}
+	client.Cleaner.Add(eventingAPIGroup, eventingAPIVersion, "brokers", namespace, name)
+	return nil
+}
+
+// CreateBrokers will create a list of Brokers.
+func (client *Client) CreateBrokers(names []string, provisionerName string) error {
+	for _, name := range names {
+		if err := client.CreateBroker(name, provisionerName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CreateTrigger will create a Trigger.
+// TODO(Fredy-Z): do not directly use TriggerFilter defined in eventing API
```

Comment:
This comment is now meaningless, I just deleted it.

The Options pattern fits really well, I've applied it on a couple of creation methods.

I did not separate the object construction and creation from user-side, since it would require one extra line of code. Now function definitions are like `CreateSubscription(name, channelName string, options ...func(*v1alpha1.Subscription))`. I think it can also provide flexibility as the code grows.